resources/js/components/TextTruncate.vue

```diff
@@ -0,0 +1,41 @@
+<template>
+  <div
+    v-b-tooltip.hover
+    :title='this.$slots.default[0].text'
+    class='text-ellipsis'
+    ref="overflow"
+    :disabled="disabled"
+  >
+    <slot></slot>
+  </div>
+</template>
+
+<script>
+export default {
+  data () {
+    return {
+      disabled: false
+    };
+  },
+  mounted () {
+    this.checkOverflow();
+    this.$nextTick(() => {
+      window.addEventListener('resize', this.checkOverflow);
+    });
+  },
+
+  beforeDestroy () {
+    window.removeEventListener('resize', this.checkOverflow);
+  },
+
+  methods: {
+    /**
+     * Enable tooltip if text overflows and is truncated
+     */
+    checkOverflow () {
+      this.disabled = this.$refs.overflow.offsetWidth >= this.$refs.overflow.scrollWidth;
```

Comment:
Please add a test that if the text gets truncated the tooltip gets shown on hover!

Reply:
This also doesn't seem to work. If the text does not get truncated the title gets also shown on hover but the title isn't styled.
![Bildschirmfoto von 2021-01-20 19-41-28](https://user-images.githubusercontent.com/6422711/105220426-fbd11c80-5b57-11eb-8bc3-2c719e403f61.png)

Reply:
I fixed the bug, but don't know how to implement a test for it. I tried but the element width is always 0, so the tooltip is never shown.

Reply:
What about a fixed width and max-widht: 0?

Reply:
Apparently we can't simulate it with real css properties, see https://github.com/jsdom/jsdom#unimplemented-parts-of-the-web-platform. I now mocked the properties for the test